mdata/schema.go

```diff
@@ -0,0 +1,68 @@
+package mdata
+
+import (
+	"github.com/raintank/metrictank/conf"
+	"github.com/raintank/metrictank/util"
+)
+
+// TTLs returns a slice of all TTL's seen amongst all archives of all schemas
+func TTLs() []uint32 {
+	ttls := make(map[uint32]struct{})
+	for _, s := range Schemas {
+		for _, r := range s.Retentions {
+			ttls[uint32(r.MaxRetention())] = struct{}{}
+		}
+	}
+	for _, r := range conf.DefaultSchema.Retentions {
+		ttls[uint32(r.MaxRetention())] = struct{}{}
+	}
+	var ttlSlice []uint32
+	for ttl := range ttls {
+		ttlSlice = append(ttlSlice, ttl)
+	}
+	return ttlSlice
+}
+
+// MaxChunkSpan returns the largest chunkspan seen amongst all archives of all schemas
+func MaxChunkSpan() uint32 {
+	max := uint32(0)
+	for _, s := range Schemas {
+		for _, r := range s.Retentions {
```

Comment:
Aren't the schemas ordered and the validation ensures that the last one always has the longest retention? Then you could just take the last one here